From 6d822e8208b3a50e431d5bdff5ac324f8f05e8ec Mon Sep 17 00:00:00 2001
From: Ovidiu Panait <ovidiu.panait@windriver.com>
Date: Thu, 14 Oct 2021 16:09:48 +0000
Subject: [PATCH 3/3] ice: fix build errors when CONFIG_XDP_SOCKETS=y

The ice ethernet driver in LTS19 was updated to the 5.12 codebase in
LINUXEXEC-10558 and adjusted to match various 5.2 APIs. When the kernel is
configured with CONFIG_XDP_SOCKETS=y, the build fails with the following errors:

1)
drivers/net/ethernet/intel/ice/ice_xsk.c:936:14: error: too few arguments to function 'ice_alloc_rx_bufs_zc'
|   936 |   failure = !ice_alloc_rx_bufs_zc(rx_ring, cleaned_count);
|       |

This was introduced by the backport of upstream commit [1]. Current codebase
still makes use of the ice_alloc_rx_bufs_fast_zc() interface that was later
removed, so use this instead of the ice_alloc_rx_bufs_zc() call introduced
by [1].

2)
drivers/net/ethernet/intel/ice/ice_xsk.c:542:20: error: 'struct ice_rxq_stats' has no member named 'page_reuse_count'
|   542 |   rx_ring->rx_stats.page_reuse_count++;
|       |

In current codebase, ice_rxq_stats structure has no page_reuse_count member, so
remove this call.

[1] https://github.com/torvalds/linux/commit/5bb0c4b5eb61d939fed0b27d11fb91fb85769c9a

Signed-off-by: Ovidiu Panait <ovidiu.panait@windriver.com>
---
 drivers/net/ethernet/intel/ice/ice_xsk.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/intel/ice/ice_xsk.c b/drivers/net/ethernet/intel/ice/ice_xsk.c
index 72bb44f987ea..fc8e58c0c4d6 100644
--- a/drivers/net/ethernet/intel/ice/ice_xsk.c
+++ b/drivers/net/ethernet/intel/ice/ice_xsk.c
@@ -539,7 +539,6 @@ ice_alloc_buf_fast_zc(struct ice_ring *rx_ring, struct ice_rx_buf *rx_buf)
 	u64 handle, hr;
 
 	if (addr) {
-		rx_ring->rx_stats.page_reuse_count++;
 		return true;
 	}
 
@@ -933,7 +932,7 @@ int ice_clean_rx_irq_zc(struct ice_ring *rx_ring, int budget)
 	}
 
 	if (cleaned_count >= ICE_RX_BUF_WRITE)
-		failure = !ice_alloc_rx_bufs_zc(rx_ring, cleaned_count);
+		failure = !ice_alloc_rx_bufs_fast_zc(rx_ring, cleaned_count);
 
 	ice_finalize_xdp_rx(rx_ring, xdp_xmit);
 	ice_update_rx_ring_stats(rx_ring, total_rx_packets, total_rx_bytes);
-- 
2.31.1

